util/chplenv/chpl_jemalloc.py

```diff
@@ -33,29 +67,39 @@ def get_uniq_cfg_path():
 # Instead of libtool or pkg-config, jemalloc uses a jemalloc-config script to
 # determine dependencies/link args . It's located in the bin directory
 @memoize
-def get_jemalloc_config_file():
-    install_path = third_party_utils.get_cfg_install_path('jemalloc')
+def get_jemalloc_config_file(target='target'):
+    install_path = third_party_utils.get_cfg_install_path('jemalloc', host_or_target=target)
     config_file = os.path.join(install_path, 'bin', 'jemalloc-config')
     return config_file
 
 
 @memoize
-def get_link_args(target_mem):
-    if target_mem == 'bundled':
-        jemalloc_config = get_jemalloc_config_file()
+def get_link_args(target, mem_val):
+    if mem_val == 'bundled':
+        jemalloc_config = get_jemalloc_config_file(target)
         libs = ['-ljemalloc']
+        # should this be an error if we can't find it?
         if os.access(jemalloc_config, os.X_OK):
+            jemalloc_libdir = run_command([jemalloc_config, '--libdir']).strip()
+            libs.append(f'-L{jemalloc_libdir}')
```

Comment:
Why is `--libdir` needed now? I thought we got our `-L` from other parts of our scripts.

Reply:
I think I threw it in there thinking it was needed when using a system install, if eg the directory is not on the lib path. But it does build without it on Mac where this would come into play

Reply:
I remember now! I did need that to build bundled. The last commit for this PR is a bit nonsensical actually.
There isn't a bunch of precedent for link args for `chpl`, only llvm, which sets its link args in the Makefile calling into `llvm-config`. I'm inclined to leave this in as it is, especially if we like the idea of `CHPL_MAKE_THIRD_PARTY_HOST_LINK_ARGS`. The counterpart `CHPL_MAKE_THIRD_PARTY_LINK_ARGS` does also include some `-L` for eg `hwloc`.

Reply:
And thinking about this again, it seems a bit unnecessary to call `jemalloc-config --libdir` when we could easily construct the right lib path with `os.path.join(install_path, 'lib')`. Does that sound good?

Reply:
Ah, makes sense. Yeah, doing `os.path.join(install_path, 'lib')` would match how we get the `-L` for other packages in `default_get_link_args`

Reply:
Fixed in https://github.com/chapel-lang/chapel/pull/18627/files#diff-c6c2f75f0251acf7b7553a7c96e1fddd18db10e53554259d05c990c3229b6a10R78-R84